ovirt_imageio/_internal/backends/http.py

```diff
@@ -416,7 +423,7 @@ def _optimize_connection(self, unix_socket):
             self._con = con
 
     def _clone_connection(self):
-        if isinstance(self._con, HTTPSConnection):
+        if isinstance(self._con, (HTTPSConnection, HTTPConnection)):
             return self._create_tcp_connection()
         else:
             return self._create_unix_connection(self.server_address)
```

Comment:
We can simplify the check like this:

    if isinstance(self._con, UnixHTTPConnection):
        return self._create_unix_connection(self.server_address)
    else:
        return self._create_tcp_connection()